src/NServiceBus.Transport.Bridge/EndpointRegistry.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NServiceBus;
+using NServiceBus.Raw;
+using NServiceBus.Transport;
+
+class EndpointRegistry : IEndpointRegistry
+{
+    public void RegisterDispatcher(
+        BridgeEndpoint endpoint,
+        string targetTransportName,
+        IRawEndpoint proxy)
+    {
+        registrations.Add(new ProxyRegistration
+        {
+            Endpoint = endpoint,
+            TranportName = targetTransportName,
+            RawEndpoint = proxy
+        });
+
+        addressMappings[endpoint.QueueAddress] = proxy.ToTransportAddress(new QueueAddress(endpoint.Name));
+    }
+
+    public void ApplyMappings(IReadOnlyCollection<BridgeTransportConfiguration> transportConfigurations)
+    {
+        foreach (var registration in registrations)
+        {
+            // target transport is the transport where this endpoint is actually running
+            var targetTransport = transportConfigurations.Single(t => t.Endpoints.Any(e => e.Name == registration.Endpoint.Name));
+
+            // just pick the first proxy that is running on the target transport since
+            // we just need to be able to send messages to that transport
+            var proxyEndpoint = registrations
+                .First(r => r.TranportName == targetTransport.Name)
+                .RawEndpoint;
+
+            targetEndpointDispatchers[registration.Endpoint.Name] = new TargetEndpointDispatcher(
+                targetTransport.Name,
+                proxyEndpoint,
+                registration.Endpoint.QueueAddress);
+        }
+    }
+
+    public TargetEndpointDispatcher GetTargetEndpointDispatcher(string sourceEndpointName)
+    {
+        return targetEndpointDispatchers[sourceEndpointName];
```

Comment:
Would it not be possible that here we get a message for an endpoint that we are unaware about? Then this would crash with "key not found" or something similar. If this is the case, then it would make sense try TryGetValue this and throw with a more descriptive message?

Reply:
Agreed, fixed now